qa/tasks/cephfs/kernel_mount.py

```diff
@@ -349,8 +354,17 @@ def get_osd_epoch(self):
         return epoch, barrier
 
     def get_op_read_count(self):
-        buf = self.read_debug_file("metrics/size")
-        if buf is None:
-            return 0
-        else:
-            return int(re.findall(r'read.*', buf)[0].split()[1])
+        stdout = StringIO()
+        stderr = StringIO()
+        try:
+            path = os.path.join(self._get_debug_dir(), "metrics/size")
+            self.run_shell(f"sudo stat {path}", stdout=stdout,
+                           stderr=stderr, cwd=None)
+            buf = self.read_debug_file("metrics/size")
+        except CommandFailedError:
+            if 'no such file or directory' in stderr.getvalue().lower() \
+                    or 'not a directory' in stderr.getvalue().lower():
+                buf = self.read_debug_file("metrics")
```

Comment:
@dparmar18 What happens if read_debug_file() return ENOENT or ENOTDIR? Looks like this will try to parse the return value the return below?

Reply:
It's under the except block, It should return with some error like `During handling of the above exception, another exception occurred:`? Also there could be only 2 cases, either the read_count will be a sub-directory(`metrics/size`) or a file(`metrics`), I don't think it should throw ENOENT or ENOTDIR here.

Reply:
What happens when "metrics" file is absent? Seems like read_debug_file() will return ENOENT...

Reply:
Is it possible that the read_counts are stored in file/dir other than `metrics` or `metrics/size`?

Reply:
> Is it possible that the read_counts are stored in file/dir other than `metrics` or `metrics/size`?

Nope. I'm worried about kernels where you don't have either files. Is that case handled?

Reply:
Oh, I didn't knew that. In the teuthology runs, I didn't encounter any such kernel. But if this needs to kept in check, I can add a try except inside the except block to handle this case.

Reply:
For most kernels, this should be fine. Just to be extra cautious and catch issues.

Reply:
Okay, I'll add the check for that.

Reply:
@vshankar I've added the condition in commit [a4979225d03b6cd699a2940e98feeb94da7a78a8](https://github.com/ceph/ceph/pull/46363/commits/a4979225d03b6cd699a2940e98feeb94da7a78a8), please have a look at it and if it's good to go I'll squash it with the 1st commit.